mvlearn/factorization/ajive.py

```diff
@@ -0,0 +1,921 @@
+            else:  # if user inputs rank list for individual matrices
+                rank = self.indiv_ranks[bn]
+                if rank == 0:
+                    U, D, V = None, None, None
+                else:
+                    U, D, V = svd_wrapper(X_orthog, rank)
+
+            # projecting X columns onto orthogonal complement of joint_scores
+
+            if self.store_full:
+                if rank == 0:
+                    I_mat = np.zeros(shape=blocks[bn].shape)
+                else:
+                    I_mat = np.array(np.dot(U, np.dot(np.diag(D), V.T)))
+            else:
+                I_mat = None  # Kill I matrix to save memory
+
+            block_specific[bn]['individual'] = {'full': I_mat,
+                                                'scores': U,
+                                                'svals': D,
+                                                'loadings': V,
+                                                'rank': rank}
+
+            # Getting the noise matrix, E
+            if self.store_full and not issparse(X):
+                E = X - (J + I_mat)
+            else:
+                E = None
+            block_specific[bn]['noise'] = E
+
+        # save view specific estimates
+        self.blocks = {}
+
+        # Stores info for easy information checking
+
+        for bn in block_specific.keys():
+            bs_dict = block_specific[bn]
+            self.blocks[bn] = \
+                ViewSpecificResults(joint=bs_dict['joint'],
+                                    individual=bs_dict['individual'],
+                                    noise=bs_dict['noise'],
+                                    block_name=bn,
+                                    obs_names=obs_names,
+                                    var_names=var_names[bn],
+                                    m=self.centers_[bn],
+                                    shape=blocks[bn].shape)
+
+        return self
+
+    @property
+    def is_fit(self):
+        if hasattr(self, "blocks"):
+            return True
+        else:
+            return False
+
+    @property
+    def block_names(self):
+        r"""
+        Returns
+        -------
+
+        block_names: list
+            The names of the views.
+
+        """
+        if self.is_fit:
+            return list(self.blocks.keys())
+        else:
+            return None
+
+    def predict(self):
+        r"""
+        Returns
+        -------
+
+        full: dict of dict of np.arrays
+            The joint, individual, and noise full estimates for each block.
+
+        """
+        full = {}
+        for bn in self.block_names:
+            full[bn] = {'joint': self.blocks[bn].joint.full_,
+                        'individual': self.blocks[bn].individual.full_,
+                        'noise': self.blocks[bn].noise_}
+
+        return full
+
+    def results_dict(self):
+        r"""
+
+        Returns
+        -------
+
+        results: dict of dict of dict of np.arrays
```

Comment:
Don't really like the idea of returning a dictionary. Would prefer a list of the three but maybe something I'll look at in terms of a larger reworking of AJIVE

Reply:
I think I'm going to change .predict() to return a list of 3. I think the results_dict should stay bc there's a decent amount of other data that is returnable (scores, svals, etc.) and it's nicely formatted

Reply:
I changed it to a list but upon and tried it but when I ended up changing it, things got a bit confusing given you have a list of lists (if you want to find the individual matrix of the third view you would have to type full[2][1] which takes a decent amount of thought). I think a potential solution could be returning a pd.DataFrame. What do you think about that?

Reply:
that being said, it isn't hard to change it to return the list of lists for the .predict() function (or even have a boolean that returns one or the other)